### PR TITLE
Add missing sparse grid headers

### PR DESCRIPTION
**Pull request by streeve**
Missing in top level header and in CMake

### CODE REVIEW

File: cajita/src/CMakeLists.txt [hunk 1]
```diff
@@ -32,11 +31,14 @@ set(HEADERS_PUBLIC
   Cajita_ParticleGridDistributor.hpp
   Cajita_Partitioner.hpp
   Cajita_ReferenceStructuredSolver.hpp
+  Cajita_SparseArray.hpp
+  Cajita_SparseDimPartitioner.hpp
+  Cajita_SparseIndexSpace.hpp
+  Cajita_SparseLocalGrid.hpp
+  Cajita_SparseLocalGrid_impl.hpp
```
Comment (abisner) [inline cajita/src/CMakeLists.txt:38]: Did you mean to include the `_impl` header?

File: cajita/src/Cajita.hpp [hunk 2]
```diff
@@ -34,8 +34,11 @@
 #include <Cajita_ParticleGridDistributor.hpp>
 #include <Cajita_Partitioner.hpp>
 #include <Cajita_ReferenceStructuredSolver.hpp>
+#include <Cajita_SparseArray.hpp>
 #include <Cajita_SparseDimPartitioner.hpp>
 #include <Cajita_SparseIndexSpace.hpp>
+#include <Cajita_SparseLocalGrid.hpp>
+#include <Cajita_SparseLocalGrid_impl.hpp>
```
Comment (abisner) [inline cajita/src/Cajita.hpp:41]: Same as above.

File: cajita/src/CMakeLists.txt (hunk 1 above)
Comment (streeve) [reply cajita/src/CMakeLists.txt:38]: Yes, we need to install

File: cajita/src/Cajita.hpp (hunk 2 above)
Comment (streeve) [reply cajita/src/Cajita.hpp]: Removed

**Review by abisner — APPROVED** [review]

**Comment by streeve** [comment]
HIP CI not available - unrelated to this PR

**Pull request merged**